### PR TITLE
perf(collaboration): support collaboration in v3

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

**Review by nliu-ms — DISMISSED** [review]

File: packages/fx-core/src/core/collaborator.ts [hunk 1]
```diff
@@ -124,6 +143,97 @@ export class CollaborationUtil {
       isAdministrator,
     };
   }
+
+  static async loadDotEnvFile(
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/src/core/collaborator.ts:147]: Will update later after multi-env is ready

File: packages/fx-core/src/core/collaborator.ts [hunk 2]
```diff
@@ -312,21 +437,27 @@ export async function checkPermission(
...
-      message: stateResult.message,
-    });
   }
   const userInfo = result.value as AppUser;
 
   if (inputs.platform === Platform.CLI) {
-    const aadAppTenantId = envInfo.state[ComponentNames.AppManifest]?.tenantId;
+    // TODO: get tenant id from .env
+    const aadAppTenantId = isApiV3Enabled()
+      ? process.env[CollaborationConstants.TeamsAppTenantIdEnv]
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/fx-core/src/core/collaborator.ts:459]: Will update this part by using tenant id from m365 account in another pr.

File: packages/fx-core/src/core/collaborator.ts [hunk 3]
```diff
@@ -124,6 +143,97 @@ export class CollaborationUtil {
...
+        new UserError(
+          SolutionSource,
+          SolutionError.FailedToLoadDotEnvFile,
+          getLocalizedString("core.collaboration.error.failedToLoadDotEnvFile", error?.message)
+        )
+      );
+    }
+  }
+
+  // Priority parameter > dotenv > env
```
Comment (SLdragon) [inline packages/fx-core/src/core/collaborator.ts:173]: From the implementation, seems priority is env > dotenv > parameter ?

Comment (KennethBWSong) [reply packages/fx-core/src/core/collaborator.ts:173]: In the implementation, to support scenario that user may have one id in env and another in dotenv, we will continue try to read id if not all two ids are provided. If app id exists in parameter, value in env/dotenv will not replace the value.

**Review by SLdragon — DISMISSED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**